src/asm/lexer.c

```diff
@@ -870,16 +872,18 @@ static void shiftChar(void)
 
 	lexerState->macroArgScanDistance--;
 
+restart:
 	if (lexerState->expansions) {
 		/* Advance within the current expansion */
-		assert(lexerState->expansions->offset < lexerState->expansions->size);
+		assert(lexerState->expansions->offset <= lexerState->expansions->size);
 		lexerState->expansions->offset++;
-		if (lexerState->expansions->offset == lexerState->expansions->size) {
+		if (lexerState->expansions->offset > lexerState->expansions->size) {
 			/* When an expansion is done, free it and move up to its parent */
```

Comment:
```suggestion
			/* When moving one past an expansion's end, free it and move up to its parent */
```

Reply:
Actually, this should say "try the movement again with its parent", or something to that effect. That would explain the `goto restart;`.